### PR TITLE
Adding new relationship class for relating two objects

### PR DESCRIPTION
**Pull request by halloffame**
Adding a relationship class that will eventually replace relates_object_tags.

Breaking the relationship out into it's own class give us better performance and more option such as sort and other relationship metadata. It also lets us see when an object is referenced from another object.

### CODE REVIEW

**Comment by rift137** [comment]
Is there a related story?

File: lib/buweb/relationship.rb [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+        a.class.to_s <=> b.class.to_s
+      end
+    end
+  end
+
+  # Validate objects are sorted before saved. This should happen automatically
+  # if you use the `new_from_objects` method to create a new relationship.
+  # This will also ensure that object_a and object_b aren't the same object.
+  def validate_order
+    if a_type > b_type || (a_type == b_type && a_id > b_id)
```
Comment (rift137) [inline lib/buweb/relationship.rb]: should we validate that they are not trying to relate an object to itself? 
a_type == b_type && a_id == b_id

**Comment by halloffame** [comment]
Don't merge this yet

**Comment by rift137** [comment]
:peach:

**Pull request merged**